### PR TITLE
Macro for Inlining SVG Files

### PR DESCRIPTION
**Pull request by jakobvogel**
> [OX-9299](https://scireum.myjetbrains.com/youtrack/issue/OX-9299)

Or combine with `inlineResource`?

### CODE REVIEW

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/pasta/noodle/macros/InlineSvgMacro.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            return factory.newDocumentBuilder().parse(stream);
+        } catch (Exception exception) {
+            throw Exceptions.handle(exception);
+        }
+    }
+
+    private String stringifyElement(Element element) {
+        try (StringWriter writer = new StringWriter()) {
+            TransformerFactory factory = TransformerFactory.newInstance();
```
Comment (idlira) [inline src/main/java/sirius/pasta/noodle/macros/InlineSvgMacro.java:108]: Can we strip this variable and just use `TransformerFactory.newInstance().newTransformer()` below? Or maybe even better, lazy-load it one inside the part?

**Review by idlira — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/pasta/noodle/macros/InlineSvgMacro.java (hunk 1 above)
Comment (jakobvogel) [reply src/main/java/sirius/pasta/noodle/macros/InlineSvgMacro.java:108]: This is pretty much the standard approach we use elsewhere, too, for instance when parsing a SAML response. I am not sure whether it is worth the effort to optimise away this single variable, considering that we are loading files from disk and sending them away over the network. (And I would expect the JIT to do that automatically anyway?)

Comment (idlira) [reply src/main/java/sirius/pasta/noodle/macros/InlineSvgMacro.java:108]: But each call will create a new factory, that could be done once as long as the part lives.
Something like that would improve it a bit, especially if this macro is called lots of times?
```
TransformerFactory factory = null;
function fetchTransformerFactory() {
  if (factory == null) {
    this.factory = TransformerFactory.newInstance();
  }
  return factory;
}
```

Comment (sabieber) [reply src/main/java/sirius/pasta/noodle/macros/InlineSvgMacro.java:108]: My 2 ct: Most of the time the argument passed to this macro will be constant (as in a string with the path to the SVG) in which case I would guess Tagliatelle can optimize caching and wont invoke the macro all of the time in those places.

Comment (andyHa) [reply src/main/java/sirius/pasta/noodle/macros/InlineSvgMacro.java:108]: I understand the concerns, but given the caching/constant nature of most calls, I'd go with the current implementation in the name of simplicity ;-)

**Review by andyHa — APPROVED** [review]

**Pull request merged**